docs/quickstart.rst

```diff
@@ -0,0 +1,553 @@
+
+First, we search through the table of identities for a matching ``IdentityInfo.identifier`` (passed as an argument). If a matching identity is not found, we check if there is already such a user in the database, we create a new user if needed and assign a new identity to that user object. Then, we save the user object (new or matched) in the Flask session.
+
+
+
+
+If ``multipass.identity_handler`` decorated method doesn't return anything, 
+``handle_auth_success`` returns ``redirect_success`` which redirects to URL stored in ``session`` as ``'_multipass_next_url'``.
+
+.. _external_logout:
+
+External provider logout
+~~~~~~~~~~~~~~~~
+
+``multipass.logout`` should be called by your application upon logout request, passing it the url to redirect to after logout and optionally a flag to clear session.
+
+.. code-block:: python
+
+	@app.route('/logout')
+	def logout():
+	    return multipass.logout(url_for('index'), clear_session=True)
+
+The ``logout`` method then calls  ``process_logout`` on provider which name was stored in ``session`` as ``'_multipass_login_provider'`` upon login.
+
+In ``process_logout`` method the provider can implement some provider-specific actions such as sending a logout notification to the provider or redirecting to a SSO logout page. The ``return_url`` from argument can be passed further if the external provider allows to specify the URL to redirect to after logging out.
+
+Notice that in our example we are using ``OAuthAuthProvider`` which has no 	``process_logout`` method implemented. Therefore we are passing ``'true'`` for ``clear_session`` to remove ``'user_id'`` that we saved in ``session`` earlier and log out the user in this way. 
+
+If there is no provider specified in ``'_multipass_login_provider'`` the ``logout`` method redirects straight to the  ``return_url`` 
+
+.. _local_providers:
+
+Local providers
+********************
+
+Configuration
+~~~~~~~~~~~~~~~~
+
+This section shows an example of a configuration for an application using a local provider. If you wish to use both external and local providers, don't hesitate to specify both local and external providers in the same configuration and just follow our guide also on :ref:`external_providers`
+
+In this example ``'test_auth_provider'`` is a dummy local authentication provider, it's linked to the ``'test_identity_provider'`` as specified in ``MULTIPASS_PROVIDER_MAP``. Specifying ``'type'`` as ``'static'`` links those providers to our ``StaticAuthProvider`` and ``StaticIdentityProvider`` example classes(More on those classes later).
+
+In ``identities`` settings of ``'test_auth_provider'`` we specify key-value pairs of username (Pig) and password (pig123), those are used for authentication by Multipass. In this example, the usernames are used as unique identifier for users. In ``identities`` settings of ``'test_identity_provider'`` we assign info keys dictionary to usernames. We also need to specify these keys in ``MULTIPASS_IDENTITY_INFO_KEYS``.
```

Comment:
I'd say "in the ``identities`` setting of ``test_auth_provider``". I'd also make the username (Pig) and password either italic or code style.